api.go

```diff
@@ -72,6 +76,46 @@ type FileMatch struct {
 	Version string
 }
 
+// ChunkMatch is a set of non-overlapping matches within a contiguous range of
+// lines in the file.
+type ChunkMatch struct {
+	// Content is a contiguous range of complete lines that fully contains Ranges.
+	Content []byte
+	// ContentStart is the location (inclusive) of the beginning of content
+	// relative to the beginning of the file. It will always be at the
+	// beginning of a line (Column will always be 1).
+	ContentStart Location
+
+	// Ranges is a set of matching ranges within this chunk. Each range is relative
+	// to the beginning of the file (not the beginning of Content).
+	Ranges []Range
+
+	// FileName indicates whether this match is a match on the file name, in
+	// which case Content will contain the file name.
+	FileName bool
+
+	Score      float64
+	DebugScore string
+}
+
+type Range struct {
+	// The inclusive beginning of the range.
+	Start Location
+	// The exclusive end of the range.
+	End Location
+
+	SymbolInfo *Symbol
+}
+
+type Location struct {
+	// 0-based byte offset from the beginning of the file
+	ByteOffset int
+	// 1-based line number from the beginning of the file
+	LineNumber int
+	// 1-based column number (in runes) from the beginning of line
+	Column int
+}
```

Comment:
WDYT of making all these ints `uint32`? Zoekt has an invariant that the size of all content in a shard is less than uint32. This allows tighter packing. In this case the size of this struct (when aligned) will shrink to 2 * 64bit instead of 3 * 64bit for example.

Reply:
Done in [b48078c](https://github.com/sourcegraph/zoekt/pull/380/commits/b48078cd1dde96694195049a25be43bd3f02c3e8)